tfdiags/doc.go

```diff
@@ -0,0 +1,16 @@
+// Package tfdiags is a utility package for representing errors and
```

Comment:
Minor nit, it's a `terraform` package already, do we need `tf` in the name?

Reply:
I originally had this called `diags` but then ran into a collision because of the idiomatic usage involving a variable called `diags`:

```go
    var diags diags.Diagnostics
```

So the "tf" prefix here is just to avoid that name collision, while also keeping the name short since it'll end up being used in a lot of different places where a long name would likely feel "noisy".

For example, the following felt too "heavy":

```go
    var diags diagnostics.Diagnostics
```

With this name, it ends up still being terse but not colliding with the variable name:

```go
    var diags tfdiags.Diagnostics
```

Other suggestions/ideas are welcome!

Reply:
Fair enough. I definitely prefer it over having to rename in the import! :+1: